src/core/public/chrome/utils.ts

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { AppCategory } from 'opensearch-dashboards/public';
+import { groupBy, sortBy } from 'lodash';
+import { ChromeNavLink } from './nav_links';
+import { ChromeRegistrationNavLink } from './nav_group';
+
+export const LinkItemType = {
+  LINK: 'link',
+  CATEGORY: 'category',
+  PARENT_LINK: 'parentLink',
+} as const;
+
+export type LinkItem = { order?: number } & (
+  | { itemType: 'link'; link: ChromeNavLink & ChromeRegistrationNavLink }
+  | { itemType: 'parentLink'; link?: ChromeNavLink & ChromeRegistrationNavLink; links: LinkItem[] }
+  | { itemType: 'category'; category?: AppCategory; links?: LinkItem[] }
+);
+
+export function getAllCategories(allCategorizedLinks: Record<string, ChromeNavLink[]>) {
+  const allCategories = {} as Record<string, AppCategory | undefined>;
+
+  for (const [key, value] of Object.entries(allCategorizedLinks)) {
+    allCategories[key] = value[0].category;
+  }
+
+  return allCategories;
+}
+
+export function fulfillRegistrationLinksToChromeNavLinks(
+  registerNavLinks: ChromeRegistrationNavLink[],
+  navLinks: ChromeNavLink[]
+): Array<ChromeNavLink & ChromeRegistrationNavLink> {
+  const allExistingNavLinkId = navLinks.map((link) => link.id);
+  return (
+    registerNavLinks
+      ?.filter((navLink) => allExistingNavLinkId.includes(navLink.id))
```

Comment:
Could `registerNavLinks` be undefined? Maybe optional chain is unnecessary, or `registerNavLinks` should be defined as optional.

Reply:
You are right, remove the unnecessary optional chain.